Logger/BaseLogger.cs

```diff
@@ -1,7 +1,9 @@
-﻿namespace Logger
-{
-    public abstract class BaseLogger
-    {
-        public abstract void Log(LogLevel logLevel, string message);
-    }
-}
+﻿namespace Logger
+{
+    public abstract class BaseLogger
+    {
+        public abstract void Log(LogLevel logLevel, string message);
+        
+        public abstract string ClassName { get; set; }

```

Comment:
```suggestion
        public string? ClassName { get; set; }
```
No need to make this abstract, its functionality isn't expected to change depending on where it is defined.